### PR TITLE
feat: Annotation API changes to support Geospatial (Map) annotations

### LINKED ISSUE #5646: Cancel any outstanding requests for annotations when navigatating away from a Notebook
**Is your feature request related to a problem? Please describe.**
Notebooks can generate a lot of requests in order to discover annotations that might link to notebook entries. If these requests are still outstanding at the time that a user navigates away from a notebook they can block the application until the requests resolve.

**Describe the solution you'd like**
Navigating away from a notebook should cancel any outstanding annotation requests.

### PR DESCRIPTION
**Pull request by ozyx**
Closes #5646 VIPEROMCT-308 

### Describe your changes:

- Modifies `getAnnotations()` to allow it to take an `abortSignal`
  - Abort pending annotations requests for Notebooks during the `beforeDestroy()` hook
  - Abort pending annotations requests for Plots during the `beforeDestroy()` hook
- Adds target comparators to the Annotation API, allowing plugins to define custom `targetDetails` comparators for specific `annotationType`s. 
  - This is useful for Map annotations, whose `targetDetails` may contain additional information specific to the feature itself, but we're only interested in matching on the `featureId`. All `targetDetails` comparisons fall back to a deep equality check if there are no registered comparators or if no comparators evaluate to true.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/annotation/AnnotationAPI.js [hunk 1]
```diff
@@ -32,13 +32,15 @@ import _ from 'lodash';
...
+ * @property {String} MAP The map annotation type
  */
 const ANNOTATION_TYPES = Object.freeze({
   NOTEBOOK: 'NOTEBOOK',
   GEOSPATIAL: 'GEOSPATIAL',
   PIXEL_SPATIAL: 'PIXEL_SPATIAL',
   TEMPORAL: 'TEMPORAL',
-  PLOT_SPATIAL: 'PLOT_SPATIAL'
+  PLOT_SPATIAL: 'PLOT_SPATIAL',
+  MAP: 'MAP'
```
Comment (scottbell) [inline src/api/annotation/AnnotationAPI.js:43]: I think the point of `GEOSPATIAL` was the same as `MAP`, so we should pick one or the other.

File: src/api/annotation/AnnotationAPI.js [hunk 2]
```diff
@@ -246,15 +248,16 @@ export default class AnnotationAPI extends EventEmitter {
...
    */
-  async getAnnotations(domainObjectIdentifier) {
+  async getAnnotations(domainObjectIdentifier, abortSignal = null) {
     const keyStringQuery = this.openmct.objects.makeKeyString(domainObjectIdentifier);
     const searchResults = (
       await Promise.all(
         this.openmct.objects.search(
           keyStringQuery,
-          null,
+          abortSignal,
```
Comment (scottbell) [inline src/api/annotation/AnnotationAPI.js:260]: Nice!

File: src/plugins/inspectorViews/annotations/AnnotationsInspectorView.vue [hunk 3]
```diff
@@ -191,8 +192,14 @@ export default {
     },
     async loadAnnotationForTargetObject(target) {
       const targetID = this.openmct.objects.makeKeyString(target.identifier);
+      if (this.abortController !== null) {
```
Comment (scottbell) [inline src/plugins/inspectorViews/annotations/AnnotationsInspectorView.vue:195]: We should add a comment here that (I think) we're aborting if the user changed target options.

**Review by scottbell — COMMENTED** [review]
@ozyx What's the best way to test this? I've throttled my network with CouchDB to very slow, but I'm not seeing canceled network requests when navigating away from an object (e.g., a notebook).

https://github.com/nasa/openmct/assets/9853862/4789a6de-9a41-41cd-9704-b682dd79a617

File: src/api/annotation/AnnotationAPI.js (hunk 1 above)
Comment (ozyx) [reply src/api/annotation/AnnotationAPI.js:43]: Ah, good to know! I initially had those marked as `GEOSPATIAL` type, but I wasn't sure if we were planning on using these for anything else, so I added in the `MAP` type along with the special handling in `AnnotationSearchResult`. I'll switch it back.

**Comment by ozyx** [comment]
> @ozyx What's the best way to test this? I've throttled my network with CouchDB to very slow, but I'm not seeing canceled network requests when navigating away from an object (e.g., a notebook).
> 
>  Screen.Recording.2023-06-01.at.3.49.02.PM.mov

@scottbell Yeah, those requests can resolve pretty quickly. A good way I found to test is to create 3 or 4 notebooks and add them to a display layout. Navigate to the display layout, and once you see the `_find` requests pop up in the network inspector, navigate away and you should see them abort.

**Comment by scottbell** [comment]
> > @ozyx What's the best way to test this? I've throttled my network with CouchDB to very slow, but I'm not seeing canceled network requests when navigating away from an object (e.g., a notebook).
> > Screen.Recording.2023-06-01.at.3.49.02.PM.mov
> 
> @scottbell Yeah, those requests can resolve pretty quickly. A good way I found to test is to create 3 or 4 notebooks and add them to a display layout. Navigate to the display layout, and once you see the `_find` requests pop up in the network inspector, navigate away and you should see them abort.

Hmm, still having trouble:

https://github.com/nasa/openmct/assets/9853862/175293a8-b103-41cb-8951-668d91648898

Is there a part of OpenMCT we can navigate to that doesn't hit CouchDB to render, but still destroys the component?

**Review by scottbell — APPROVED** [review]
LGTM

**Pull request merged**